components/crypto/src/lib.rs

```diff
@@ -461,13 +482,22 @@ implement_index_traits! {SecretKey}
 implement_index_traits! {Seed}
 implement_index_traits! {Signature}
 
+/// Pair of matching secret and public keys.
+///
+/// Prefer using this struct to `(PublicKey, SecretKey)`, since it asserts in compile time
```

Comment:
In compile time? How?

Reply:
Perhaps not the best phrasing; what I meant is that public and secret keys in a `KeyPair` instance are guaranteed to match. Indeed, the `from_keys` constructor asserts this, and other constructors guarantee this by design.

Reply:
I'd remove "at compile time" part: code with an incorrect keypair will compile, but will panic in the runtime because of the assertion failure.